### PR TITLE
Dynamic ISF

### PR DESCRIPTION
**Pull request by MilosKozak**

### CODE REVIEW

File: app/src/main/assets/OpenAPSSMB/determine-basal.js [hunk 1]
```diff
@@ -920,10 +920,10 @@ var determine_basal = function determine_basal(glucose_status, currenttemp, iob_
                 rT.reason += ", but Min. Delta " + minDelta.toFixed(2) + " > Exp. Delta " + convert_bg(expectedDelta, profile);
             }
             if (currenttemp.duration > 15 && (round_basal(basal, profile) === round_basal(currenttemp.rate, profile))) {
-                rT.reason += ", temp " + currenttemp.rate + " ~ req " + round(basal, 2) + "U/hr. ";
+                rT.reason += ", temp " + currenttemp.rate + " ~ req " + basal + "U/hr. ";
```
Comment (paravoid) [inline app/src/main/assets/OpenAPSSMB/determine-basal.js:923]: This (and a couple references below) seem like they are reverting ca0478572aaccb660f47d001a9f53b6fc614e341 - is that intended?

**Comment by ga-zelle** [comment]
Can you please give this a name like DynamicISF or similar instead of AutoISF? AutoISF has become a recognised name by dozens of users and implements a completely different approach.

File: app/src/main/assets/OpenAPSSMBDynamicISF/determine-basal.js [hunk 2]
```diff
@@ -0,0 +1,1231 @@
...
+        if (now < 1){
+            now = 1;}
+        else {
+            console.error("Time now is "+now+"; ");
+        }
+    if (meal_data.TDDAIMI7){
+        var tdd7 = meal_data.TDDAIMI7;
+        }
+    else{
+    var tdd7 = ((basal * 12)*100)/21;
```
Comment (Andries-Smit) [inline app/src/main/assets/OpenAPSSMBDynamicISF/determine-basal.js:291]: Basal is potential adjusted by sensitivity, https://github.com/nightscout/AndroidAPS/pull/1325/files#diff-5ad799bd5b1bbda834b7197b4122b17cc17e9a8cc62e4442b15eff7a94e4f994R221
Is this by intent?

File: app/src/main/assets/OpenAPSSMBDynamicISF/determine-basal.js [hunk 3]
```diff
@@ -0,0 +1,1231 @@
...
+    var minDelta = Math.min(glucose_status.delta, glucose_status.short_avgdelta);
+    var minAvgDelta = Math.min(glucose_status.short_avgdelta, glucose_status.long_avgdelta);
+    var maxDelta = Math.max(glucose_status.delta, glucose_status.short_avgdelta, glucose_status.long_avgdelta);
+
+    var profile_sens = round(profile.sens,1)
+    var sens = profile.sens;
+
+    var now = new Date().getHours();
+        if (now < 1){
+            now = 1;}
```
Comment (Andries-Smit) [inline app/src/main/assets/OpenAPSSMBDynamicISF/determine-basal.js:283]: Now we got double period 1.
Should this be
``` js
var now = new Date().getHours() + 1;
```

File: app/src/main/assets/OpenAPSSMBDynamicISF/determine-basal.js [hunk 4]
```diff
@@ -0,0 +1,1231 @@
...
+        }
+    else{
+    var tdd7 = ((basal * 12)*100)/21;
+    }
+        var tdd_pump_now = meal_data.TDDPUMP;
+        var tdd_pump = ( tdd_pump_now / (now / 24));
+        var TDD = (tdd7 * 0.4) + (tdd_pump * 0.6);
+        console.error("Pump extrapolated TDD = "+tdd_pump+"; ");
+        //if (tdd7 > 0){
+        if ( tdd_pump > tdd7 && now < 5 || now < 7 && TDD < ( 0.8 * tdd7 ) ){
```
Comment (Andries-Smit) [inline app/src/main/assets/OpenAPSSMBDynamicISF/determine-basal.js:298]: Can we comment why this `now < 5 || now < 7`?
If the time of the day is a factor why? and what about time zone shifting, daylight saving time, profile shifts adjustments?

**Review by Andries-Smit — COMMENTED** [review]
Review JavaScript

**Pull request merged**